src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
+        {
+            BatchSharedKeyCredential creds = new BatchSharedKeyCredential(ClientUnitTestCommon.DummyAccountName, ClientUnitTestCommon.DummyAccountKey);
+
+            HttpRequestMessage message = new HttpRequestMessage(httpMethod, url);
+            message.Headers.Add("client-request-id", Guid.NewGuid().ToString());
+
+            await creds.ProcessHttpRequestAsync(message, CancellationToken.None);
+            Assert.NotNull(message.Headers.Authorization);
+
+            void AssertLambda(RequestContext ctx)
+            {
+                if (httpMethod == HttpMethod.Head || httpMethod == HttpMethod.Get)
```

Comment:
Agree with basically all you said.

> So we'd either need a way of extracting the StringToSign from our BatchSharedKeyCredential (do we expose this?) or else compute the hash again on the server side and validate that they match.

Yes, I have another item to enable the end to end integration tests to run in both netcore and netframework, but in order to do that there is more work required. So instead of doing that for this issue (since it's customer impacting), I added a targeted unit test. We will circle back around and enable the end to end tests to run against both, at which point we should get the validation we seek (since the E2E tests use all the different verbs)